### ISSUE TITLE
Manage our tools' dependencies in a structured way

### ISSUE THREAD

**Issue by mlavacca** [kubernetes-sigs/gateway-api#3318]
**What would you like to be added**:

We have many dependencies in our hack script that often do not get updated. A non-exhaustive list follows:
- https://github.com/kubernetes-sigs/gateway-api/blob/cb846ce842f47074781de7ebfbb1080f3c471f2f/hack/verify-crds-kind.sh#L48-L49
- https://github.com/kubernetes-sigs/gateway-api/blob/cb846ce842f47074781de7ebfbb1080f3c471f2f/hack/verify-golint.sh#L21-L22
- https://github.com/kubernetes-sigs/gateway-api/blob/cb846ce842f47074781de7ebfbb1080f3c471f2f/hack/verify-yamllint.sh#L21-L22

We need to discover all these dependencies and set up a way to either
- centralize them in a unique place easier and quicker to manage and update, or
- use an automatic tool to automatically bump them, such as [renovate](https://docs.renovatebot.com/)

**Comment by pmalek** [comment]
> centralize them in a unique place easier and quicker to manage and update, or
> use an automatic tool to automatically bump them, such as [renovate](https://docs.renovatebot.com/)

Why not both?

I'd suggest the following approach:

- put tools names and versions in 1 structured file ( e.g. a yaml called `.tools_versions.yaml`)
- use those from `Makefile` using a tool like `yq`
- add renovate annotations to the file so that these tools are kept up to date

In order to keep a unified interface for installing and running the tools we can leverage something like [`mise`](https://github.com/jdx/mise/) which will handle installing these tools' binaries. The only requirement for end user will be to have `mise` installed. Everything else will be handled by the tooling.

We follow a similar approach in 
https://github.com/Kong/gateway-operator/blob/d401801b2b98bd771b841dc9af49073c1722cf1d/.tools_versions.yaml and it has worked well for us. I'd be happy to contribute this change if there's a consensus that this is the correct way forward.

**Comment by mlavacca** [comment]
> > centralize them in a unique place easier and quicker to manage and update, or
> > use an automatic tool to automatically bump them, such as [renovate](https://docs.renovatebot.com/)
> 
> Why not both?
> 
> I'd suggest the following approach:
> 
>     * put tools names and versions in 1 structured file ( e.g. a yaml called `.tools_versions.yaml`)
> 
>     * use those from `Makefile` using a tool like `yq`
> 
>     * add renovate annotations to the file so that these tools are kept up to date
> 
> 
> In order to keep a unified interface for installing and running the tools we can leverage something like [`mise`](https://github.com/jdx/mise/) which will handle installing these tools' binaries. The only requirement for end user will be to have `mise` installed. Everything else will be handled by the tooling.
> 
> We follow a similar approach in https://github.com/Kong/gateway-operator/blob/d401801b2b98bd771b841dc9af49073c1722cf1d/.tools_versions.yaml and it has worked well for us. I'd be happy to contribute this change if there's a consensus that this is the correct way forward.

Yep, this approach makes a lot of sense to me. Just want to check with the other maintainers to see if they are on board

/cc @shaneutt @robscott @youngnick

**Comment by youngnick** [comment]
Agreed, it seems a little Rube-Goldberg-y to me, but seems worth trying as well.
